components/status_dropdown/status_dropdown.jsx

```diff
@@ -105,6 +149,112 @@ export default class StatusDropdown extends React.PureComponent {
             </FormattedMessage>
         );
     }
+    handleClearStatus = (e) => {
+        e.stopPropagation();
+        this.props.actions.unsetCustomStatus();
+    };
+
+    onToggle = (open) => {
+        this.props.actions.setStatusDropdown(open);
+    }
+
+    renderCustomStatus = () => {
+        if (!this.props.isCustomStatusEnabled) {
+            return null;
+        }
+        let customStatusText = localizeMessage('status_dropdown.set_custom', 'Set a Custom Status');
+        let customStatusEmoji = <EmojiIcon className={'custom-status-emoji'}/>;
+        const customStatus = this.props.customStatus;
+        const isStatusSet = customStatus && (customStatus.text || customStatus.emoji);
+        if (isStatusSet) {
+            customStatusEmoji = (
+                <span className='d-flex'>
+                    <CustomStatusEmoji
+                        showTooltip={false}
+                    />
+                </span>
+            );
+            customStatusText = customStatus.text;
+        }
+
+        let customStatusTextComponent = (
+            <span
+                className='custom_status__text'
+                ref={(element) => {
+                    this.customStatusTextRef = element;
+                    this.showCustomStatusTextTooltip();
+                }}
+            >
+                <Markdown
+                    message={customStatusText}
+                    enableFormatting={true}
+                />
+            </span>
+        );
+
+        if (isStatusSet && this.state.showCustomStatusTooltip) {
+            customStatusTextComponent = (
+                <OverlayTrigger
+                    delayShow={Constants.OVERLAY_TIME_DELAY}
+                    placement='bottom'
+                    overlay={
+                        <Tooltip id='custom-status-text-tooltip'>
+                            <Markdown
+                                message={customStatusText}
+                                enableFormatting={true}
+                            />
+                        </Tooltip>
+                    }
+                >
+                    {customStatusTextComponent}
+                </OverlayTrigger>
+            );
+        }
```

Comment:
this whole block seems very redundant over the different files. 

I would suggest extracting it into a helper function or component that can be imported instead.